Add release script

m-mujica opened the pull request:
Closes #1059

matthewp commented:
@m-mujica if you could, please add this step to the wiki page: https://github.com/stealjs/steal/wiki/Release-Process

after merging

m-mujica commented:
Oh shoot, I missed the "after merging" part and edited already :|

m-mujica commented:
I removed the step about updating the changelog since we don't have one and added the step to run `gh-release` after creating the tag.

```
Test IE8 and IE9 to ensure no breakage.
Bump the bower.json and package.json version numbers.
Do a build (will set the version correctly).
Commit to master.
Create a tag.
Publish.
Create Github release (run npm run gh-release)
Generate new docs.
```
